### PR TITLE
feat: add swagger to backend

### LINKED ISSUE #38: feat: add swagger docs to backend API
Add swagger documentation to the backend api for easy API documentation, as well as enforce proper http exceptions for every endpoint

### PR DESCRIPTION
**Pull request by aidanaden**
closes #38 and #42

### CODE REVIEW

File: backend/src/main.ts [hunk 1]
```diff
@@ -21,6 +23,15 @@ async function bootstrap() {
   // const cookieSecret = app.get(ConfigService).getOrThrow("COOKIE_SECRET");
   const port = app.get(ConfigService).get("PORT");
 
+  patchNestJsSwagger();
+  const swaggerConfig = new DocumentBuilder()
+    .setTitle("G420 PeerPussies API")
+    .setDescription("The REST interface for querying the G420 PeerPussies API")
```
Comment (yusufaine) [inline backend/src/main.ts:29]: KIV group name

**Review by yusufaine — APPROVED** [review]
Tested locally with redis DB running to simulate the endpoints. 
Not able to test happy path as verification is not within this PR, but error messages do line up.

LGTML 🌮

**Review by blaketingyu — APPROVED** [review]
LGTM!

File: backend/src/main.ts (hunk 1 above)
Comment (blaketingyu) [reply backend/src/main.ts:29]: HAHAHHA

**Pull request merged**